agent/consul/internal_endpoint.go

```diff
@@ -205,6 +205,87 @@ func (m *Internal) GatewayServiceDump(args *structs.ServiceSpecificRequest, repl
 	return err
 }
 
+// Match returns the set of intentions that match the given source/destination.
+func (m *Internal) GatewayIntentions(args *structs.IntentionQueryRequest, reply *structs.IndexedIntentions) error {
```

Comment:
Is this supposed to be a modified mirror of `Intention.Match`? If so I think the part below about setting empty namespaces to `"default"` is slightly wrong, and should be replaced by lines like this from the `intention_endpoint.go:Match`:

```
	// Get the ACL token for the request for the checks below.
	var entMeta structs.EnterpriseMeta
	authz, err := s.srv.ResolveTokenAndDefaultMeta(args.Token, &entMeta, nil)
	if err != nil {
		return err
	}

...later...

	// Finish defaulting the namespace fields.
		if args.Match.Entries[0].Namespace == "" {
			args.Match.Entries[0].Namespace = entMeta.NamespaceOrDefault()
		}
		if err := s.srv.validateEnterpriseIntentionNamespace(args.Match.Entries[0].Namespace, true); err != nil {
			return fmt.Errorf("Invalid match entry namespace %q: %v",
				args.Match.Entries[0].Namespace, err)
		}


```

Reply:
Fixed